bundles/gcp/src/main/java/org/apache/gravitino/gcs/fs/GCSFileSystemProvider.java

```diff
@@ -43,6 +45,19 @@ public FileSystem getFileSystem(Path path, Map<String, String> config) throws IO
     Configuration configuration = new Configuration();
     FileSystemUtils.toHadoopConfigMap(config, GRAVITINO_KEY_TO_GCS_HADOOP_KEY)
         .forEach(configuration::set);
+
+    if (config.containsKey(GravitinoVirtualFileSystemConfiguration.FS_GRAVITINO_SERVER_URI_KEY)) {
```

Comment:
I don't think checking `FS_GRAVITINO_SERVER_URI_KEY ` is a proper way. whether we set `GravitinoGCSCredentialProvider` depends on the `credentials` or `credential providers` for the fileset.

Reply:
Please see below